src/emqx_delayed_publish.erl

```diff
@@ -41,27 +41,21 @@
 
 -spec(load() -> ok).
 load() ->
-    emqx:hook('message.publish', {?MODULE, on_message_publish, []}).
+    emqx:hook('message.publish', {?MODULE, on_message_publish, []}),
+    emqx:hook('session.resumed', {?MODULE, on_session_resumed, []}).
 
-on_message_publish(Msg = #message{id = Id, topic = <<"$delayed/", Topic/binary>>, timestamp = Ts}) ->
+on_message_publish(Msg = #message{topic = <<"$delayed/", Topic/binary>>}) ->
```

Comment:
worth introducing a macro for `$delayed` like `$queue` and `$share`, see `emqx.hrl`